doxia-site-renderer/src/main/java/org/apache/maven/doxia/siterenderer/ParserConfigurator.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.maven.doxia.siterenderer;
+
+import java.nio.file.Path;
+
+import org.apache.maven.doxia.parser.Parser;
+
+/**
+ * Interface which allows to configure a particular {@link Parser} before it is being used.
+ */
+public interface ParserConfigurator {
+
+    /**
+     * Configures the given parser, which is afterwards used to parse the given source file.
+     * @param parserId the parser id
+     * @param filePath the absolute path of the file to parse
+     * @param parser the parser to configure
+     * @return {@code true} if the parser has been configured, otherwise {@code false}
+     */
+    boolean configure(String parserId, Path filePath, Parser parser);
```

Comment:
Last solution was to limit the config to the parser id and not fine grained to files...